app/services/publishers/vacancies/vacancy_step_process.rb

```diff
@@ -24,6 +24,14 @@ def validatable_steps
     steps - %i[documents review]
   end
 
+  def previous_visited_step
+    return steps.last if vacancy.completed_steps.include?(steps.last.to_s) && first_of_group?
```

Comment:
I don't think this is right, it does a different thing compared to what the `back_path` does right now.

`review` doesn't get added to `completed_steps` until all steps are valid: https://github.com/DFE-Digital/teaching-vacancies/blob/master/app/controllers/publishers/vacancies_controller.rb#L29

So you might come from the review page because you're editing an incomplete draft, but with this logic the "Back" link would take you to the previous step rather than the review page.